### PR TITLE
Fix OCSP length check and make extension length checking more robust (1.0.2)

### PR DESCRIPTION
**Pull request by mattcaswell**
This is basically the same fix as #1713 but for the 1.0.2 branch. There were a number of conflicts which I had to resolve so I could not do a clean cherry-pick.

### CODE REVIEW

**Comment by mattcaswell** [comment]
Updated as per #1713 to use a macro.

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -1274,10 +1276,8 @@ unsigned char *ssl_add_clienthello_tlsext(SSL *s, unsigned char *buf,
          * 2 for hostname length
          * + hostname length
          */
-
-        if ((lenmax = limit - ret - 9) < 0
-            || (size_str =
-                strlen(s->tlsext_hostname)) > (unsigned long)lenmax)
+        size_str = strlen(s->tlsext_hostname);
+        if (CHECKLEN(ret, 9 + size_str, limit))
```
Comment (richsalz) [inline ssl/t1_lib.c]: I think ```CHECKLEN``` is a not-great name, ```TOOBIG``` or similar is better.  See https://www.lysator.liu.se/c/pikestyle.html on function names :)  But that's up to you.

**Comment by mattcaswell** [comment]
@richsalz Can you also approve the #1713 version of this?

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**